sdk/iothub/azure-resourcemanager-iothub/src/test/java/com/azure/resourcemanager/iothub/generated/IotHubManagerTests.java

```diff
@@ -0,0 +1,176 @@
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        iotHubManager = IotHubManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testIotHubDescription() {
+        IotHubDescription iotHubDescription = null;
+        try {
+            String descriptionName = "iotHub" + randomPadding();
+            // embedmeStart
+            iotHubDescription = iotHubManager.iotHubResources()
+                .define(descriptionName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroupName)
+                .withSku(new IotHubSkuInfo().withName(IotHubSku.F1).withCapacity(1L))
+                .withIdentity(new ArmIdentity().withType(ResourceIdentityType.NONE))
+                .withProperties(
+                    new IotHubProperties()
+                        .withIpFilterRules(Arrays.asList())
+                        .withEventHubEndpoints(mapOf("events",
+                            new EventHubProperties()
+                                .withRetentionTimeInDays(1L)
+                                .withPartitionCount(2)))
+                        .withRouting(new RoutingProperties()
+                            .withEndpoints(new RoutingEndpoints()
+                                .withEventHubs(Arrays.asList())
+                                .withServiceBusQueues(Arrays.asList())
+                                .withServiceBusTopics(Arrays.asList())
+                                .withStorageContainers(Arrays.asList())
+                                .withCosmosDBSqlCollections(Arrays.asList()))
+                            .withRoutes(Arrays.asList())
```

Comment:
what happens if we don't set these empty lists?

we don't want to show user these empty list, which appear to be meaningless.